app/assets/src/components/Samples.jsx

```diff
@@ -938,53 +923,31 @@ class Samples extends React.Component {
     return !this.state.columnsShown.includes(column);
   }
 
-  initializeSelectAll() {
-    // select all checkboxes
-    var that = this;
-    $(".checkAll").click(function(e) {
-      var checked = e.currentTarget.checked;
-      $(".checkbox:enabled").prop("checked", checked);
-      that.setState({
-        allChecked: checked
-      });
-      that.fetchAllSelectedIds(checked);
-    });
-  }
-
-  checkTheRightBoxes() {
-    var that = this;
-    $(".checkbox:enabled").each((id, element) => {
-      let sample_id = element.getAttribute("data-sample-id");
-      const sampleList = that.state.selectedSampleIds;
-      if (!sample_id) {
-        return;
+  fetchAllSelectedIds(e) {
+    let sampleList = this.state.selectedSampleIds;
+    const checked = e.target.checked;
+    const allSamples = this.state.allSamples;
+    for (let idx in allSamples) {
```

Comment:
You can use of in the for statement to get the actual values instead of just the idx. 

```
for (let sample of allSamples)
```

Reply:
ah thx